File: solutions/typescript/resistor-color-duo/1/resistor-color-duo.ts
export function decodedValue(colors: string[]): number {
  const number1 = colorList.indexOf(colors[0]) * 10;
  const number2 = colorList.indexOf(colors[1]);

  return number1 + number2;
}

const colorList: string[] = [
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white' 
];

File: solutions/python/resistor-color-duo/1/resistor_color_duo.py
resistor_values_by_color = {
    "black": 0,
    "brown": 1,
    "red": 2,
    "orange": 3,
    "yellow": 4,
    "green": 5,
    "blue": 6,
    "violet": 7,
    "grey": 8,
    "white": 9
    }

def value(colors):
    color1, color2, *rest = colors
    value1 = resistor_values_by_color[color1]
    value2 = resistor_values_by_color[color2]
    return (value1 * 10) + (value2)
